app/Repositories/Three/PostRepository.php

```diff
@@ -0,0 +1,254 @@
+     * @param Registrar $registrar
+     */
+    public function __construct(AWS $aws, Registrar $registrar)
+    {
+        $this->aws = $aws;
+        $this->registrar = $registrar;
+    }
+
+    /**
+     * Find a post by post_id and return associated signup and tags.
+     *
+     * @param int $id
+     * @return \Rogue\Models\Post
+     */
+    public function find($id)
+    {
+        return Post::with('signup', 'tags')->findOrFail($id);
+    }
+
+    /**
+     * Create a Post.
+     *
+     * @param  array $data
+     * @param  int $signupId
+     * @return \Rogue\Models\Post|null
+     */
+    public function create(array $data, $signupId)
+    {
+        if (isset($data['file'])) {
+            // Auto-orient the photo by default based on exif data.
+            $image = Image::make($data['file'])->orientate();
+
+            $fileUrl = $this->aws->storeImage((string) $image->encode('data-url'), $signupId);
+        } else {
+            $fileUrl = 'default';
+        }
+
+        $signup = Signup::find($signupId);
+
+        // Create a post.
+        $post = new Post([
+            'signup_id' => $signup->id,
+            'northstar_id' => auth()->id(),
+            'campaign_id' => $signup->campaign_id,
+            'quantity' => $data['quantity'],
+            'url' => $fileUrl,
+            'caption' => $data['caption'],
+            'status' => isset($data['status']) ? $data['status'] : 'pending',
+            'source' => token()->client(),
+            'remote_addr' => isset($data['remote_addr']) ? $data['remote_addr'] : null,
+        ]);
+
+        $post->save();
+
+        // Edit the image if there is one
+        if (isset($data['file'])) {
+            $this->crop($data, $post->id);
+        }
+
+        return $post;
+    }
+
+    /**
+     * Update an existing Post and Signup.
+     *
+     * @param \Rogue\Models\Signup $signup
+     * @param array $data
+     *
+     * @return Signup|Post
+     */
+    public function update($signup, $data)
+    {
+        if (array_key_exists('updated_at', $data)) {
+            // Should only update why_participated and timestamps on the signup
+            $signupFields = [
+                'why_participated' => isset($data['why_participated']) ? $data['why_participated'] : null,
+                'updated_at' => $data['updated_at'],
+                'created_at' => array_key_exists('created_at', $data) ? $data['created_at'] : null,
+            ];
+
+            // Only update if the key is set (is not null).
+            $nonNullArrayKeys = array_filter($signupFields);
+            $arrayKeysToUpdate = array_keys($nonNullArrayKeys);
+
+            $signup->fill(array_only($data, $arrayKeysToUpdate));
+            $signup->save(['timestamps' => false]);
+
+            $event = $signup->events->last();
+            $event->created_at = $data['updated_at'];
+            $event->updated_at = $data['updated_at'];
+            $event->save(['timestamps' => false]);
+        } else {
```

Comment:
@katiecrane I'm pretty sure we can, but just want to double check - can I delete all of the above (lines 111-130)?

Reply:
@chloealee for V3 YES 🔥